src/test/java/reposense/parser/ArgsParserTest.java

```diff
@@ -63,20 +62,20 @@ public void parse_configFileOnly_success() throws ParseException, IOException, j
         String input = String.format("-config %s", CONFIG_FILE_ABSOLUTE);
         CliArguments cliArguments = ArgsParser.parse(translateCommandline(input));
         Files.isSameFile(expected, cliArguments.getConfigFilePath());
-        //Optional arguments have default values
+        // Optional arguments have default values
         Assert.assertEquals(Optional.empty(), cliArguments.getSinceDate());
         Assert.assertEquals(Optional.empty(), cliArguments.getUntilDate());
-        //Checks if generate output folder is in yyyy-MM-dd-HH-mm-ss format
-        Constants.DEFAULT_REPORT_NAME_FORMAT.parse(cliArguments.getOutputFilePath().getFileName().toString());
+        // Checks if output folder generated is in correct format
+        ArgsParser.DEFAULT_REPORT_NAME_FORMAT.parse(cliArguments.getOutputFilePath().getFileName().toString());
```

Comment:
https://docs.oracle.com/javase/9/docs/api/java/text/SimpleDateFormat.html#parse-java.lang.String-java.text.ParsePosition-

I think you need to additionally check that the result is not null.